### PR TITLE
Improve an error message when privileged containers are disallowed on the cluster

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it**:

At present when user creates privileged pod and creation of privileged containers disallowed globally by a system administrator (kubelet and api-server were running with `--allow-privileged=false`), user will get the following error message:
```console
$ kubectl create -f nginx.pod 
The Pod "nginx" is invalid: spec.containers[0].securityContext.privileged: Forbidden: disallowed by policy
```
"Disallowed by policy" may give a wrong assumption to a user that creation of privileged containers disallowed by [`PodSecurityPolicy`](http://kubernetes.io/docs/user-guide/pod-security-policy/) while it's not.

This commit improves error message and tries to point user to the right direction:
```console
$ kubectl create -f nginx.pod 
The Pod "nginx" is invalid: spec.containers[0].securityContext.privileged: Forbidden: privileged containers are disallowed on this cluster by a system administrator
```

**Release note**:

```release-note
NONE
```

PTAL @pweil-

### CODE REVIEW

**Comment by pweil-** [comment]
@k8s-bot ok to test

**Review by pweil- — APPROVED** [review]
Change is fine with me.  I hope we can remove this flag at some point

**Comment by thockin** [comment]
@k8s-bot ok to test

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -3692,7 +3692,8 @@ func ValidateSecurityContext(sc *api.SecurityContext, fldPath *field.Path) field
 
 	if sc.Privileged != nil {
 		if *sc.Privileged && !capabilities.Get().AllowPrivileged {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("privileged"), "disallowed by policy"))
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("privileged"),
+				"privileged containers are disallowed on this cluster by a system administrator"))
```
Comment (thockin) [inline pkg/api/validation/validation.go]: In general we don't repeat the field name in the error message.  What if this was just "disallowed by administrative policy" or "disallowed by cluster policy" ?

Comment (php-coder) [inline pkg/api/validation/validation.go]: I don't mind to modify the message but  the main intention of this change was to remove word "policy" from the message. When I saw this I thought about PodSecurityPolicy but this is completely unrelated. Ideally it should tell that administrator didn't pass `--allow-privileged` option to kubelet and api-server but probably you wouldn't agree with such message too?

**Comment by thockin** [comment]
But it *is* a policy decision.  I would not reference a specific flag name.
 "disallowed by cluster policy" seems to eliminate confusion, no?  Maybe I
am too close to it..

**Comment by php-coder** [comment]
>  "disallowed by cluster policy" seems to eliminate confusion, no?

@thockin ok, let's use this message. It's still better than the current. I've updated PR.

**Comment by php-coder** [comment]
@k8s-bot unit test this https://github.com/kubernetes/kubernetes/issues/39604
